src/stream/ascii_reader.rs

```diff
@@ -0,0 +1,398 @@
+        }
+    }
+
+    fn line_comment(&mut self) -> Result<(), Error> {
+        // Consumes up to the end of the line.
+        // There's no error in this a line comment can reach the EOF and there's
+        // no forbidden chars in comments.
+        while {
+            match self.peeked_char {
+                Some(c) => c != b'\n',
+                None => false,
+            }
+        } {
+            let _ = self.advance()?;
+        }
+
+        Ok(())
+    }
+
+    fn block_comment(&mut self) -> Result<(), Error> {
+        let mut latest_consume = b' ';
+        while {
+            latest_consume != b'*'
+                || match self.advance()? {
+                    Some(c) => c != b'/',
+                    None => false,
+                }
+        } {
+            latest_consume = self.advance()?.unwrap_or(b' ');
+        }
+
+        Ok(())
+    }
+
+    /// Returns:
+    /// - Some(string) if '/' was the first character of a string
+    /// - None if '/' was the beginning of a comment.
+    fn potential_comment(&mut self) -> Result<Option<Event>, Error> {
+        match self.peeked_char {
+            Some(c) => match c {
+                b'/' => self.line_comment().map(|_| None),
+                b'*' => self.block_comment().map(|_| None),
+                _ => self.unquoted_string_literal(c),
+            },
+            // EOF
+            None => Err(self.error(ErrorKind::IncompleteComment)),
+        }
+    }
+
+    /// Consumes the reader until it finds a valid Event
+    /// Possible events for Ascii plists:
+    ///  - StartArray(Option<u64>),
+    ///  - StartDictionary(Option<u64>),
+    ///  - EndCollection,
+    ///  - Data(Vec<u8>),
+    fn read_next(&mut self) -> Result<Option<Event>, Error> {
+        while let Some(c) = self.advance()? {
+            match c {
+                // Single char tokens
+                b'(' => return Ok(Some(Event::StartArray(None))),
+                b')' => return Ok(Some(Event::EndCollection)),
+                b'{' => return Ok(Some(Event::StartDictionary(None))),
+                b'}' => return Ok(Some(Event::EndCollection)),
+                b'"' => return self.quoted_string_literal(),
+                b'/' => {
+                    match self.potential_comment() {
+                        Ok(Some(event)) => return Ok(Some(event)),
+                        Ok(None) => { /* Comment has been consumed */ }
+                        Err(e) => return Err(e),
+                    }
+                }
+                b',' | b';' | b'=' => { /* consume these without emitting anything */ }
+                b' ' | b'\r' | b'\t' | b'\n' => { /* whitespace is not significant */ }
+                _ => return self.unquoted_string_literal(c),
+            }
+        }
+
+        return Ok(None);
+    }
+}
+
+impl<R: Read> Iterator for AsciiReader<R> {
+    type Item = Result<Event, Error>;
+
+    fn next(&mut self) -> Option<Result<Event, Error>> {
+        match self.read_next() {
```

Comment:
This can be just `self.read_next().transpose()`. Don't worry that this will fail on the 1.32.0 Travis builder.